### DISCUSSION TITLE
Meter is not being discovered on MODBUS TCP

### DISCUSSION THREAD

**Discussion by mayberryjp** [wlcrs/huawei_solar#536] [Connecting to the inverter]
Hi,

I have a smart meter it's connected and I can't see data on COM1. I can also see smart meter stats like grid power and return on the smartloggers local website. The meter is registered in the smart logger and everything looks normal. The battery and inverter are discovered fine by home assistant and the entities have data and work 100% as expected. 

The problem is - Home Assistant can't discover the smart meter and I can't get any meter entities in home assistant. 

Does anyone have any ideas? I've tried everything - removing and re-adding the meter, power cycling the smart logger, etc. Nothing is working. It seems to be a MODBUS TCP issue but i can't figure it out. 

I am connecting via MODBUS TCP directly to the smart loggers management IP. There are all kinds of settings around  logical address and communication address and slave IDs. it's not obvious to me how to figure out the meters slave ID. To discover the battery and inverter it is on slave ID 2. 

thanks

![image](https://github.com/wlcrs/huawei_solar/assets/147527091/1691a832-1831-4561-a368-eb8e8c796dc0)

**Comment by Roving-Ronin** [comment]
@mayberryjp SmartLogger is NOT a supported configuration. If you want to get this integration working, you must rewire the meter to the primary inverter, plug the sDongle into the primary inverter and connect using Modbus TCP to the IP of the sDongle (that you've also enabled modbus tcp access upon).

PS. Said as a person with 2 x SmartLogger 3000A's..... ended up having to remove the SmartLoggers and rewire my setup to use this intergration. Personal call on what you want/need more the data and ability to automate in Home Assistant or local SmartLogger vs FusionSolar in the cloud.

**Reply by mayberryjp** [reply]
I understand your answer. But, I guess I am curious - if the data is visible in the smartlogger UI, then what causes this scenario to not be supported?  I 

I am a dev so was reverse engineering the smart logger UI a bit - so maybe it's easier or better if i write an integration for my own use. The smart logger just exposes the meter information on a simple like CSV style output with a fixed URL. 

There's some hackish authentication in front of the smart logger UI, but nothing too difficult to overcome i don't think. and maybe there's even a way to turn off the authentication.

**Reply by Roving-Ronin** [reply]
@mayberryjp if your a dev and can add support for the SmartLogger into wlcrs integration, then please do ! :-)  Would be great to be able to use SmartLogger (that also supports multiple modbus tcp clients, whilst the dongle only supports a single client).

Can't remember when I had wlcrs connecting via the SmartLogger if it was seeing all the Huawei Optimizers (don't think that feature was added to wlcrs at that point), but all of the other functions to do with the inverters where working fine. Not having a LUNA battery to test that, the only issue I encountered was that the meter (DTSU-666H) was not availble/visible. 

iirc I would guess that this might be because the meter modbus register numbers are different (than when the meter is connected to the inverter).

There is a (closed) thread about SmartLogger support, https://github.com/wlcrs/huawei_solar/issues/93#issuecomment-1700332589
wlcrs dropped adding it, as he doesn't have a SmartLogger to allow testing any code changes against.  
There was a branch setup for adding this support at: https://gitlab.com/Emilv2/huawei-solar/-/tree/smartlogger-support

Happy to test if you can add this support to wlcrs, with a view to getting it pulled / merged into the wlcrs integration 👍

**Reply by mayberryjp** [reply]
OK thanks. I read through the thread. None of this is my domain of knowledge - so I have to learn about modbus, etc. I was first attempting to try and use modbuspoll and see if there are even SmartMeter registers. But, so far haven't gotten very far. At least, that seemed like a good place to start without investing too much on a failed effort. Apparently all my tinkering has resulted in my smart meter getting disconnected from my smart logger, so i need to figure that out first.
